File: claude/jsp/fiassed/src/main/java/com/loosenotes/model/Attachment.java
package com.loosenotes.model;

import java.time.LocalDateTime;

/**
 * Attachment metadata model.
 * storedFilename is a UUID-based server-generated name (F-05, Integrity).
 * originalFilename is stored only for display/download header — never used as filesystem path.
 */
public class Attachment {

    private Long id;
    private Long noteId;
    private String storedFilename;   // UUID.ext - used for filesystem path
    private String originalFilename; // User-supplied name - used only in Content-Disposition header
    private String contentType;
    private long fileSize;
    private LocalDateTime createdAt;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getNoteId() { return noteId; }
    public void setNoteId(Long noteId) { this.noteId = noteId; }

    public String getStoredFilename() { return storedFilename; }
    public void setStoredFilename(String storedFilename) { this.storedFilename = storedFilename; }

    public String getOriginalFilename() { return originalFilename; }
    public void setOriginalFilename(String originalFilename) {
        this.originalFilename = originalFilename;
    }

    public String getContentType() { return contentType; }
    public void setContentType(String contentType) { this.contentType = contentType; }

    public long getFileSize() { return fileSize; }
    public void setFileSize(long fileSize) { this.fileSize = fileSize; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: claude/aspnet/fiassed/LooseNotes/Models/Attachment.cs
namespace LooseNotes.Models;

/// <summary>
/// File attachment metadata. StoredFileName is server-generated (UUID);
/// OriginalFileName is display-only metadata, never used in file system paths (ASVS V5.3.2).
/// </summary>
public sealed class Attachment
{
    public int Id { get; set; }
    public int NoteId { get; set; }
    public string UserId { get; set; } = string.Empty;

    /// <summary>
    /// UUID-based name used for file system storage. Never derived from user input.
    /// </summary>
    public string StoredFileName { get; set; } = string.Empty;

    /// <summary>
    /// Original filename stored as metadata only. Never used to construct file paths.
    /// </summary>
    public string OriginalFileName { get; set; } = string.Empty;

    public string ContentType { get; set; } = string.Empty;
    public long FileSizeBytes { get; set; }
    public DateTime UploadedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public Note Note { get; set; } = null!;
    public ApplicationUser User { get; set; } = null!;
}
